staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -409,7 +416,29 @@ func (c *converterImpl) ToUnstructured(obj interface{}) (map[string]interface{},
 	if err != nil {
 		return nil, err
 	}
-	return *u, nil
+	return u, nil
+}
+
+// DeepCopyJSON deep copies the passed value, assuming it is a valid JSON representation i.e. only contains
+// types produced by json.Unmarshal().
+func DeepCopyJSON(x interface{}) interface{} {
```

Comment:
I feel like this name is confusing, since x is not actually json?

Reply:
I hoped the comment above clarifies why it is called that but I'm happy to rename it to anything. What would you name it?

Reply:
Do we have a better name? It's the go representation of JSON.

Reply:
Can we rename this to `deepCopyJSON` and then put a `DeepCopyJSON(x map[string]interface{}` in front of it? Then at least we get a bit of type-checking.

Reply:
I had this idea too. But then we loose the ability to pass `[]interface{}` to it or just `interface{}` that may contain any of those types. This may be useful - `json.Unmarshal()` produces something into `interface{}` and then you may want to deep copy that.

Reply:
Do we have situations where we have general JSON or only objects? I believe in Unstructured we only have objects.

Reply:
Changed the signature.